modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/resolver/ByParameterNameArgumentResolver.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.runtime.resolver;
+
+import org.mule.extension.runtime.OperationContext;
+
+/**
+ * An implementation of {@link ArgumentResolver} which resolves
+ * to a parameter value of name {@link #parameterName}
+ *
+ * @param <T> the type of the argument to be resolved
```

Comment:
For this particular implementation the generic gives no value... but it does have value for other implementations of ArgumentResolver